### DISCUSSION TITLE
swanctl configuration

### DISCUSSION THREAD

**Discussion by dhodyrahmad** [strongswan/strongswan#224] [General]
Hi, Im a newbie trying to solve some task with Strongswan. I worked on Strongswan v5.8.2 with CentOS 8 try to connected to other VPN on other company (site-to-site). I've done read some Strongswan configuration documentation. I just want to back to basic wanna know that i don't missed something. I worked using swanctl and trying to established connection to other VPN. Im following the documentation and asked before on strongswan problem page. I can telnet but i get _DUP!_ on ping, there is something wrong with my configuration?

let say:
```
----------------                                       ----------------
|            A            | --------------------- |            B           |
|   Strongswan   |                                      |   Other VPN    |
----------------                                       ----------------
IP public  : 34.101.xxx.xxx                       IP public   : 103.28.xxx.x
IP private : 10.148.x.x                              IP private  : 192.168.xx.xx
```
*NOTE: Other VPN only allow at port 9001

this is my configuration:
```
connections {
    site-to-site {
        version = 1
        remote_addrs = 103.28.xxx.x
        local_port = 500
        remote_port = 500
        proposals = aes256-sha1-modp1536
        keyingtries = 1
        rekey_time = 86400s
        local {
            auth = psk
            id = 34.101.xxx.xxx
        }
       remote {
            auth = psk
            id = 103.28.xxx.x
       }
       children {
            site-to-site {
                esp_proposals = aes128-sha1
                local_ts = 10.184.x.x/30
                remote_ts = 192.168.xx.xx/30
                life_time = 3600s
                mode = tunnel
            }
       }
   }
}

secrets {
    ike {
        secret = _My Secret PSK_
        id-1a = 34.101.xxx.xxx
        id-1b = 103.28.xxx.x
    }
}
```

did my configuration wrong? I checked the log but everythings go well. I inititate with swanctl --inititate and return some log, in the end says "initiate completed successfully"

**Comment by strongX509** [comment]
Hello,

without a strongSwan log I cannot see if you really set up a succcesful IPsec connection . That you could telnet but were not able to ping is very strange. Could you give me the destination addresses you wanted to reach via telnet and ICMP? With such scarce information it is nearly impossible to debug your problem. You also write that the VPN gateway can be only reached on port 9001 which is an OpenVPN port but you define the standard UDP 500 ports for IKE. Please explain!

Andreas

**Reply by dhodyrahmad** [reply]
Hello,

here some log when swanctl initiates:
```
[IKE] initiating Main Mode IKE_SA site-to-site[34] to 103.28.xxx.x
[ENC] generating ID_PROT request 0 [ SA V V V V V ]
[NET] sending packet: from 10.184.x.x[500] to 103.28.xxx.x[500] (184 bytes)
[NET] received packet: from 103.28.xxx.x[500] to 10.184.x.x[500] (128 bytes)
[ENC] parsed ID_PROT response 0 [ SA V V ]
[IKE] received FRAGMENTATION vendor ID
[IKE] received NAT-T (RFC 3947) vendor ID
[CFG] selected proposal: IKE:AES_CBC_256/HMAC_SHA1_96/PRF_HMAC_SHA1/MODP_1536
[ENC] generating ID_PROT request 0 [ KE No NAT-D NAT-D ]
[NET] sending packet: from 10.184.x.x[500] to 103.28.xxx.x[500] (308 bytes)
[NET] received packet: from 103.28.xxx.x[500] to 10.184.x.x[500] (296 bytes)
[ENC] parsed ID_PROT response 0 [ KE No NAT-D NAT-D ]
[IKE] local host is behind NAT, sending keep alives
[ENC] generating ID_PROT request 0 [ ID HASH N(INITIAL_CONTACT) ]
[NET] sending packet: from 10.184.x.x[4500] to 103.28.xxx.x[4500] (108 bytes)
[NET] received packet: from 103.28.xxx.x[4500] to 10.184.x.x[4500] (140 bytes)
[ENC] parsed ID_PROT response 0 [ ID HASH N((24576)) V ]
[IKE] received DPD vendor ID
[IKE] IKE_SA site-to-site[34] established between 10.184.x.x[34.101.xxx.xxx]...103.28.xxx.x[103.28.xxx.x]
[IKE] scheduling rekeying in 85278s
[IKE] maximum IKE_SA lifetime 93918s
[ENC] generating QUICK_MODE request 1702967772 [ HASH SA No ID ID ]
[NET] sending packet: from 10.184.x.x[4500] to 103.28.xxx.x[4500] (172 bytes)
[NET] received packet: from 103.28.xxx.x[4500] to 10.184.x.x[4500] (172 bytes)
[ENC] parsed QUICK_MODE response 1702967772 [ HASH SA No ID ID ]
[CFG] selected proposal: ESP:AES_CBC_128/HMAC_SHA1_96/NO_EXT_SEQ
[IKE] CHILD_SA site-to-site{57} established with SPIs c1c34eb2_i 91007ddd_o and TS 10.184.x.x/32 === 192.168.xx.xx/30
[ENC] generating QUICK_MODE request 1702967772 [ HASH ]
[NET] sending packet: from 10.184.x.x[4500] to 103.28.xxx.x[4500] (60 bytes)
initiate completed successfully
```

> Could you give me the destination addresses you wanted to reach via telnet and ICMP?

destination address 192.168.xx.xx/30

> You also write that the VPN gateway can be only reached on port 9001 which is an OpenVPN port but you define the standard UDP 500 ports for IKE. Please explain!

this makes me curious too, this is first time I using and setup VPN. I practiced to configure between strongswan and strongswan. then its little bit confusing when its come to setup to other VPN. should i give the remote_port 9001?

**Reply by strongX509** [reply]
The IKEv1 tunnel setup is indeed successful. Port 9001 is never used by IKE/IPsec. I cannot explain why your ping produces duplicate packets.

**Reply by dhodyrahmad** [reply]
> The IKEv1 tunnel setup is indeed successful. Port 9001 is never used by IKE/IPsec. I cannot explain why your ping produces duplicate packets.

so this is means every traffic through port 9001 are outside the tunnel?

**Reply by strongX509** [reply]
Port 9001 is also tunneled

**Reply by dhodyrahmad** [reply]
so, based on my configuration and my logs, its clearly fine?

**Reply by strongX509** [reply]
No traffic on port 9001 is tunneled as well.

**Reply by dhodyrahmad** [reply]
```
10:44:43.862231 IP_internal_A > IP_public_B.ipsec-nat-t: UDP-encap: ESP(spi=0xd11b6a6f,seq=0xd7), length 100
10:44:43.886639 IP IP_public_B.ipsec-nat-t >  IP_internal_A.ipsec-nat-t: UDP-encap: ESP(spi=0xc22a36b3,seq=0x2e9), length 100
10:44:43.886734 IP IP_public_B.ipsec-nat-t >  IP_internal_A.ipsec-nat-t: UDP-encap: ESP(spi=0xc22a36b3,seq=0x2ea), length 100
10:44:43.886741 IP IP_public_B.ipsec-nat-t >  IP_internal_A.ipsec-nat-t: UDP-encap: ESP(spi=0xc22a36b3,seq=0x2eb), length 100
10:44:43.886747 IP IP_public_B.ipsec-nat-t >  IP_internal_A.ipsec-nat-t: UDP-encap: ESP(spi=0xc22a36b3,seq=0x2ec), length 100
10:44:43.886752 IP IP_public_B.ipsec-nat-t >  IP_internal_A.ipsec-nat-t: UDP-encap: ESP(spi=0xc22a36b3,seq=0x2ed), length 100
10:44:43.886762 IP IP_public_B.ipsec-nat-t > IP_internal_A.ipsec-nat-t: UDP-encap: ESP(spi=0xc22a36b3,seq=0x2ee), length 100
10:44:43.886767 IP IP_public_B.ipsec-nat-t > IP_internal_A.ipsec-nat-t: UDP-encap: ESP(spi=0xc22a36b3,seq=0x2ef), length 100
```
This log are dumped from strongswan using ```tcpdump -i eth0 udp``` at server A and tested with telnet to **private IP B** and **port 9001**
